core/network/src/sync.rs

```diff
@@ -239,6 +239,13 @@ impl<B: BlockT> ChainSync<B> {
 							}
 						},
 						None => {
+							if n > As::sa(0) {
+								trace!(target:"sync", "Ancestry block not found for peer {}: block #{}", who, n);
+								let n = n - As::sa(1);
```

Comment:
`One::one()` is more idiomatic.

```suggestion
								let n = n - One::one();
```